controller/konnect/ops/ops.go

```diff
@@ -37,6 +38,35 @@ const (
 	DeleteOp Op = "delete"
 )
 
+// EntityTypeName is the type of the Konnect entity name used for distinguish operations on different types of entities in the prometheus metrics.
+type EntityTypeName string
+
+// Entity type names for Konnect entity as labels in metrics.
+const (
+	// Entity type names used in metrics.
+	// REVIEW: Should we use the path inside the API as the type names? These are not very consistent in
+	EntityTypeControlPlane         EntityTypeName = "control_planes"
+	EntityTypeService              EntityTypeName = "services"
+	EntityTypeRoute                EntityTypeName = "routes"
+	EntityTypeConsumer             EntityTypeName = "consumers"
+	EntityTypeConsumerGroup        EntityTypeName = "consumer_groups"
+	EntityTypePlugin               EntityTypeName = "plugins"
+	EntityTypeUpstream             EntityTypeName = "upstreams"
+	EntityTypeTarget               EntityTypeName = "targets"
+	EntityTypeBasicAuthCredential  EntityTypeName = "basic_auth_credentials" //nolint:gosec
+	EntityTypeAPIKeyCredential     EntityTypeName = "api_key_credentials"    //nolint:gosec
+	EntityTypeACLCredential        EntityTypeName = "acl_credentials"
+	EntityTypeJWTCredential        EntityTypeName = "jwt_credentials"
+	EntityTypeHMACCredential       EntityTypeName = "hmac_credentials"
+	EntityTypeCACertificate        EntityTypeName = "ca_certificates"
+	EntityTypeCertificate          EntityTypeName = "certificates"
+	EntityTypeSNI                  EntityTypeName = "snis"
+	EntityTypeKey                  EntityTypeName = "keys"
+	EntityTypeKeySet               EntityTypeName = "key_sets"
+	EntityTypeVault                EntityTypeName = "vaults"
+	EntityTypeDataPlaneCertificate EntityTypeName = "data_plane_certificates"
```

Comment:
All of these types should have the `GetTypeName() string` method (if that's not the case we can change the code generation in kcfg) which will give you the  kind. We can use that instead of these.

`Create` / `Update` / `Delete` can then use `e.GetTypeName()` from the `constraints.EntityType[T]` type constraint (instead of calling that in each `switch` branch.

Reply:
Done.